packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryBuffer.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+// Default duration threshold for reporting performance entries (0 means "report
+// all")
+constexpr double DEFAULT_DURATION_THRESHOLD = 0.0;
+
+/**
+ * Abstract performance entry buffer with reporting flags.
+ * Subtypes differ on how entries are stored.
+ */
+class PerformanceEntryBuffer {
+public:
+  double durationThreshold{DEFAULT_DURATION_THRESHOLD};
+  size_t droppedEntriesCount{0};
```

Comment:
Do these need to be public?

Reply:
It needs to be accessed from EntryReporter to validate if the entry should be added. I can make it protected, but then I'd need to make it a friend of `EntryReporter` which I'd avoid. As an alternative I could move the login into buffer itself, but that was just moved out of it :D

Reply:
I don't think we need durationThreshold here. We can just define it in the buffer for event entry types.

Regarding `droppedEntriesCount`, there's no reason to have them per buffer AFAIK, so we could make `add` return a boolean and then have a single count in the entry reporter, so you don't even have to iterate over each buffer. It's a bit simpler IMHO.